src/MongoDB/ReadPreference.c

```diff
@@ -436,15 +441,16 @@ static HashTable* php_phongo_readpreference_get_properties_hash(zval* object, bo
 		PHONGO_BSON_INIT_STATE(state);
 		state.map.root_type = PHONGO_TYPEMAP_NATIVE_ARRAY;
 
-		php_phongo_bson_to_zval_ex(bson_get_data(tags), tags->len, &state);
+		if (!php_phongo_bson_to_zval_ex(bson_get_data(tags), tags->len, &state)) {
+			zval_ptr_dtor(&state.zchild);
+			return props;
```

Comment:
This is analogous to `return Z_ARRVAL(retval);` in the `get_debug_info` handlers, but it doesn't stand out to me as much since we're not using a macro here.

I think one concern I had with these lines getting out of sync (and suggesting `goto`) is also that we don't really test them. Some `php_phongo_bson_to_zval_ex` calls in this PR could theoretically be tested by mocking the server to return garbage data, but I don't think we could cover everything. The original motivation of PHPC-348 was just add some defensive checks since it was obvious we weren't checking for errors (not considering how likely or unlikely those errors might be).